src/atomate2/common/jobs/anharmonicity.py

```diff
@@ -0,0 +1,733 @@
+    """
+
+    def __init__(self, largest_mode: float) -> None:
+        self.largest_mode = largest_mode
+        self.message = f"""Structure has imaginary modes: the largest optical
+                        eigenmode {largest_mode} < 0.0001"""
+        super().__init__(self.message)
+
+
+@job
+def get_phonon_supercell(
+    structure: Structure,
+    supercell_matrix: np.ndarray,
+) -> Structure:
+    """Get the phonon supercell of a structure.
+
+    Parameters
+    ----------
+    structure: Structure
+        The structure to get phonon supercell for
+    supercell_matrix: np.ndarray
+        Supercell matrix for the phonon
+
+    Returns
+    -------
+    Structure
+        The phonopy structure
+    """
+    cell = get_phonopy_structure(structure)
+    phonon = Phonopy(
+        cell,
+        supercell_matrix,
+    )
+    return get_pmg_structure(phonon.supercell)
+
+
+def get_sigma_per_site(
+    structure: Structure,
+    forces_dft: np.ndarray,
+    forces_harmonic: np.ndarray,
+) -> list[tuple]:
+    """Compute a site-resolved sigma^A.
+
+    This will calculate a sigma^A value for every
+    site in the structure.
+
+    Parameters
+    ----------
+    structure: Structure
+        The structure to use for the calculation
+    forces_dft: np.ndarray
+        DFT calculated forces
+    forces_harmonic: np.ndarray
+        Harmonic approximation of the forces
+
+    Returns
+    -------
+    list[tuple]
+        List of tuples in the form
+        ({wyckoff symbol: sites}, sigma^A) for
+        all the sites in the structure
+    """
+    # Ensure that DFT and harmonic forces are in np format
+    forces_dft = np.array(forces_dft)
+    forces_harmonic = np.array(forces_harmonic)
+
+    # Check shapes of forces
+    if len(np.shape(forces_dft)) == 2:
+        forces_dft = np.expand_dims(forces_dft, axis=0)
+        forces_harmonic = np.expand_dims(forces_harmonic, axis=0)
+
+    sg_analyzer = SpacegroupAnalyzer(structure)
+    sym_dataset = sg_analyzer.get_symmetry_dataset()
+    wycks = np.array(sym_dataset.wyckoffs)
+    sites = np.array(structure.sites)
+
+    if np.shape(forces_dft)[1] == 3 * structure.num_sites:
+        sites = sites.repeat(3)
+        wycks = wycks.repeat(3)
+
+    unique_sites = np.unique(wycks)
+    site_to_wyckoff: dict = {wyck: [] for wyck in unique_sites}
+    for idx, val in enumerate(wycks):
+        site_to_wyckoff[val].append(sites[idx])
+
+    sigma_sites = []
+    for s in unique_sites:
+        mask = wycks == s
+        f_dft = forces_dft[:, mask]
+        f_ha = forces_harmonic[:, mask]
+        f_anharm = f_dft - f_ha
+        sigma_sites.append(calc_sigma_a(f_anharm, f_dft))
+
+    return [
+        (
+            {
+                list(site_to_wyckoff.keys())[i]: site_to_wyckoff[
+                    list(site_to_wyckoff.keys())[i]
+                ]
+            },
+            sigma_sites[i],
+        )
+        for i in range(len(sigma_sites))
+    ]
+
+
+def get_sigma_per_element(
```

Comment:
@4kevinbeck5 this looks very good. 

The only comment that I have is that some of these are functions and are not really jobs. @utf should we move those somewhere else? `common/utils` would be a possibility but it might get messy.

Reply:
The rationale for making those functions rather than jobs was just to reduce the number of jobs per flow on the HPC

Reply:
This is completely fine and desired. I am just wondering if functions should be in the `jobs` folder/file.